### DISCUSSION TITLE
Demo: openNewCodeEditor not used?

### DISCUSSION THREAD

**Discussion by kaisalmen** [CodinGame/monaco-vscode-api#525] [Q&A] [answered]
Hi @CGNonofr I recognized that the `openNewCodeEditor` is not used. I am currently integrating ViewsService in a new `monaco-languageclient` demo and was confused that the function I wrote was not used. Editors are opened and seem to properly work without using the configured function. I test this with a local build of your demo and it is not used either. Do you have an idea why?

**Comment by CGNonofr** [comment, marked as answer]
The view service allows you to use any component. The demo showcases the editor part but one can chose to use only the panel component, and still be using regular editors.

So the view service override still takes the openEditor fallback for that specific case.

Looking at the code, it may also be called if for some reasons the file open was "refused" by VSCode

**Reply by CGNonofr** [reply]
That prop is optional though, you can probably ignore it

**Reply by kaisalmen** [reply]
Ok, makes sense. Thank you for explaining.
